scala/scala.bzl

```diff
@@ -410,13 +437,13 @@ def _collect_jars_from_common_ctx(ctx, extra_deps = [], extra_runtime_deps = [])
     # Get jars from deps
     auto_deps = [ctx.attr._scalalib, ctx.attr._scalareflect]
     deps_jars = _collect_jars(ctx.attr.deps + auto_deps + extra_deps)
-    (cjars, transitive_rjars, jars2labels) = (deps_jars.compile_jars, deps_jars.transitive_runtime_jars, deps_jars.jars2labels)
+    (cjars, transitive_rjars, jars2labels, transitive_cjars) = (deps_jars.compile_jars, deps_jars.transitive_runtime_jars, deps_jars.jars2labels, deps_jars.transitive_cjars)
 
     runtime_dep_jars =  _collect_jars(ctx.attr.runtime_deps + extra_runtime_deps)
     transitive_rjars += runtime_dep_jars.transitive_runtime_jars
 
     jars2labels.update(runtime_dep_jars.jars2labels)
-    return struct(compile_jars = cjars, transitive_runtime_jars = transitive_rjars, jars2labels=jars2labels)
+    return struct(compile_jars = cjars, transitive_runtime_jars = transitive_rjars, jars2labels=jars2labels, transitive_cjars = transitive_cjars)
```

Comment:
@johnynek WDYT about having `transitive_cjars` be named `compile_jars` because intent wise that's what they are. The jars which we're going to compile with. Implementation wise they are transitive but I'm not sure that's the focus.
And I'd rename the existing `compile_jars` to be `direct_jars` since that's the only reason they're still needed. For the plugin to know if the used jar is `direct` or `indirect`.
WDYT?